src/burnchains/burnchain.rs

```diff
@@ -1327,19 +1343,20 @@ impl Burnchain {
                     while let Ok(Some(burnchain_block)) = db_recv.recv() {
                         debug!("Try recv next parsed block");
 
-                        if burnchain_block.block_height() == 0 {
+                        let block_height = burnchain_block.block_height();
+                        if block_height == 0 {
                             continue;
                         }
 
                         if is_mainnet {
-                            if last_processed.block_height == STACKS_2_0_LAST_BLOCK_TO_PROCESS {
+                            if block_height == STACKS_2_0_LAST_BLOCK_TO_PROCESS {
```

Comment:
Is this line tested by any existing tests? This is the logic to stop the chain, right?

Reply:
should we remove this snippet since we know we no longer want to stop at that block height?

Reply:
I reverted these changes, and will let Jude remove this snippet when he implements the on-chain exit-at-block-height. 
In addition to the `is_mainnet` issue, there is a bug in there where instead of looking at block height of the block we just downloaded, we're using the height of the block that was initially the chain tip when the sync started.
Fixing it does not matter since this code path is dead anyway.